lightning-invoice/src/utils.rs

```diff
@@ -337,18 +337,61 @@ fn _create_invoice_from_channelmanager_and_duration_since_epoch<M: Deref, T: Der
 	network: Currency, amt_msat: Option<u64>, description: InvoiceDescription,
 	duration_since_epoch: Duration, invoice_expiry_delta_secs: u32
 ) -> Result<Invoice, SignOrCreationError<()>>
-where
-	M::Target: chain::Watch<<K::Target as KeysInterface>::Signer>,
-	T::Target: BroadcasterInterface,
-	K::Target: KeysInterface,
-	F::Target: FeeEstimator,
-	L::Target: Logger,
+	where
+		M::Target: chain::Watch<<K::Target as KeysInterface>::Signer>,
+		T::Target: BroadcasterInterface,
+		K::Target: KeysInterface,
+		F::Target: FeeEstimator,
+		L::Target: Logger,
 {
 	// `create_inbound_payment` only returns an error if the amount is greater than the total bitcoin
 	// supply.
 	let (payment_hash, payment_secret) = channelmanager
 		.create_inbound_payment(amt_msat, invoice_expiry_delta_secs)
 		.map_err(|()| SignOrCreationError::CreationError(CreationError::InvalidAmount))?;
+	_create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_hash(
+		channelmanager, keys_manager, logger, network, amt_msat, description, duration_since_epoch, invoice_expiry_delta_secs, payment_hash, payment_secret)
+}
+
+/// See [`create_invoice_from_channelmanager_and_duration_since_epoch`]
+///Function that allows to construct invoices with the given payment hash.
+///This may be useful if you're building an on-chain swap or integrating with another protocol where the payment hash is fixed outside of lightning.
```

Comment:
Nice, we're getting there now! Just some formatting (trying to stay within 100 char line length where possible, and space after `///`) and slight rewording suggestion:
```suggestion
/// This version allows for providing a custom [`PaymentHash`] for the invoice.
/// This may be useful if you're building an on-chain swap or involving another protocol where
/// the payment hash is also involved outside the scope of lightning.
```